### PR TITLE
FOGL-7074 Make storage plugin selection a choice from a drop down list

### PR DESCRIPTION
**Pull request by MarkRiddoch**
Signed-off-by: Mark Riddoch <mark@dianomic.com>

### CODE REVIEW

**Review by pintomax — APPROVED** [review]

**Review by ashish-jabble — CHANGES_REQUESTED** [review]
Now with fresh instance it's fine.
But with existing/upgrade way it's still come up with old configuration of Storage category.
Detailed testing notes are available on [FOGL-7074 JIRA](https://scaledb.atlassian.net/browse/FOGL-7074?focusedCommentId=47004&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-47004)

**Review by pintomax — DISMISSED** [review]

**Comment by ashish-jabble** [comment]
@MarkRiddoch Now things are good with above commits. But having some concerns related with readingPlugin value is empty in storage.json when upgrade And on fresh setup there are some syslog errors related with No such category found (Storage). Detailed notes are available on [JIRA](https://scaledb.atlassian.net/browse/FOGL-7074?focusedCommentId=47076&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-47076)

**Review by pintomax — APPROVED** [review]

File: C/services/storage/storage.cpp [hunk 1]
```diff
@@ -364,13 +364,19 @@ void StorageService::start(string& coreAddress, unsigned short corePort)
...
+					// Regsiter for configuration changes to our reading category category
 					ConfigHandler *configHandler = ConfigHandler::getInstance(client);
 					configHandler->registerCategory(this, conf->getName());
 				}
 			}
 		}
 
+		// Now we are running force the plugin names back to the configuration manager to
+		// make sure they match what we are running. This can be out of sync if the storage
+		// configuration cache has been manaully reset or altered while Fledge was down
```
Comment (ashish-jabble) [inline C/services/storage/storage.cpp:376]: manually

**Review by ashish-jabble — APPROVED** [review]
LGTM!

**Comment by ashish-jabble** [comment]
@MarkRiddoch on fledge reset we are not resetting `storage.json`. Shall we now?

**Comment by MarkRiddoch** [comment]
> @MarkRiddoch on fledge reset we are not resetting `storage.json`. Shall we now?

I don't think we should.

**Review by pintomax — APPROVED** [review]

**Pull request merged**